### PR TITLE
[ashell] Fix for UART not showing text properly

### PR DESCRIPTION
**Pull request by brianjjones**
Fix for #1368

Signed-off-by: Brian Jones <brian.j.jones@intel.com>

### CODE REVIEW

File: src/ashell/comms-uart.c [hunk 1]
```diff
@@ -313,16 +313,20 @@ void comms_write_buf(const char *buf, int len)
...
-    }
-    while (data_transmitted == false) {
-        ;
+    int sent = 0;
+
+    while (len > 0) {
+        data_transmitted = false;
+        sent = uart_fifo_fill(dev_upload, (const u8_t *)buf, len);
+
+        while (data_transmitted == false) {
```
Comment (grgustaf) [inline src/ashell/comms-uart.c]: I think you should only do this busy-wait if sent > 0. If it returns 0 you would presumably hang forever.

File: src/ashell/comms-uart.c [hunk 2]
```diff
@@ -313,16 +313,20 @@ void comms_write_buf(const char *buf, int len)
...
-        len -= bytes;
-        bytes = uart_fifo_fill(dev_upload, (const u8_t *)buf, len);
-    }
-    while (data_transmitted == false) {
-        ;
+    int sent = 0;
+
+    while (len > 0) {
+        data_transmitted = false;
+        sent = uart_fifo_fill(dev_upload, (const u8_t *)buf, len);
```
Comment (grgustaf) [inline src/ashell/comms-uart.c]: The docs for uart_fifo_fill seem to indicate this should only be called from an ISR, but we're calling this function all over the place I think. Maybe we need to go back to UART samples and see how it's properly used. Or the docs could be wrong; haven't looked into it myself.

**Comment by grgustaf** [comment]
+1

**Comment by brianjjones** [comment]
@grgustaf It says changes requested, but I see no change requests?  Just curious what I need to change to get this accepted.  Thanks!

**Comment by grgustaf** [comment]
I made the changes myself in a second commit and I guess was just waiting for @jimmy-huang to accept it...

**Comment by jimmy-huang** [comment]
+1

**Pull request merged**